ome/omero-iviewer
# Ol build cleanup

This PR removes all the build steps required by the google closure build of ol3-viewer and other cleanup https://trello.com/c/vyZSm0fj/15-openlayers-5-build-cleanup

Removed:

 - build.xml targets:
    - clean
    - init
    - compile
    - compile-debug
    - copy-to-lib
    - prepare-unit-tests
    - plugin (replaced with 'npm run plugin')
    - build-all
    - build-debug
    - build
- closure/calcdeps.py
- closure/compiler.jar
- ol3_viewer_prepare.sh (deleted in #218)

To test:

 - We should still be able to run the same commands:
    - ```npm run debug```
    - ```npm run prod```
    - ```npm run dev```
    - ```npm run docs``` - This includes ol3-viewer docs in /build/docs/iviewer now (no separate ol3-viewer docs)
    - ```npm run plugin``` which builds viewer plugin only (replaces ```ant build-plugin```)
        - Ideally try to ```omero config append omero.web.apps '"ol3-viewer"'``` and open ```/ol3-viewer/ID``` (see docs below)

 - Check documentation updates:
     - https://github.com/will-moore/omero-iviewer/blob/ol_build_cleanup/README.rst
     - https://github.com/will-moore/omero-iviewer/tree/ol_build_cleanup/plugin/ol3-viewer

Not yet done: ```ant unit-tests```. This will require some more discussion and bigger changes (currently investigating) so would be best to leave to another PR.
